src/engine/handles/RobotHandle.ts

```diff
@@ -14,6 +14,10 @@ export class RobotHandle extends ObjectHandle<SimRobot> {
     return this._rootObject.getAnalogInput(channel);
   }
 
+  setDigitalInput(channel: number, value: boolean): void {
```

Comment:
I like that the robot can, through a few hops (like the MechanisManager) send control inputs to the SimGripperMechanism to tell it to open or clause the gripper.

Feel free to ignore: During my initial reading of your change I was a bit confused by the name `setDigitalInput`, in particular that it controls the Gripper.  To me controlling the gripper is similar to setting the motor power on an axis - it's basically controlling an actuator.  And controlling the motors happens via `setMotorPower` - not via some set*Input(...) method.  Maybe that's just me looking into this codebase for the first time.

Reply:
I would hope that eventually there will be a top level interface for the robot. Everything (including the motors) would be accessed or controlled by four methods.:
 setDigitalInput, getDigitalInput, setAnalogInput, getAnalogInput.

That covers motors, wheels. grabbers sensors etc...

Thats is why the mechanisms are connected like this in code. its just a start obviously we should change the abstraction in the robot code too in order to support this.

Reply:
Should it be `setDigitalOutput`/`setAnalogOutput` instead then? Since you aren't actually setting a value on an input channel (that you will then read in robot code)

Reply:
The 4 methods mentioned above aren't exactly how you would do it on an actual robot. The `setDigitalOutput`/`getDigitalInput` is fine, but there are some difference between setting an analog output value (which usually outputs a variable voltage) vs setting a motor power command (which sends signals over PWM).

However I think for this use case, the simpler abstraction should be fine.

Reply:
Yes the name was indeed incorrect, i've switched input here for output.